src/texstudio.cpp

```diff
@@ -710,8 +710,21 @@ void Texstudio::setupDockWidgets()
 
     leftPanel->showWidgets();
     // restore selected view in sidepanel
-    int viewNr=configManager.getOption("GUI/sidePanel/currentPage", 0).toInt();
-    leftPanel->setCurrentWidget(leftPanel->widget(viewNr));
+    int viewNr = configManager.getOption("GUI/sidePanel/currentPage", 0).toInt();
+	int k = viewNr;
+	int i;
+	for (i = 0; i < leftPanel->widgetCount(); i++)  {
+		if (!leftPanel->hiddenWidgets().split("|").contains(leftPanel->widget(i)->property("id").toString())) {
+			if (k == 0) {
```

Comment:
1) please don't nest to many methods
leftPanel->hiddenWidgets().split("|") -> stringlist abc
def = leftPanel->widget(i)->property("id").toString()
abc contains def

the counting up and counting down of k looks strange

Reply:
so I guess you want to show the nth visible widget, so count k up from 0 and compare it to viewNr ...

Reply:
? I want to show the (viewNr+1)th visible widget (which has index viewNr). If k equals zero then the loop entered (viewNr+1) times the if and we are done. I'll reverse counting.

Tell me more about how do define abc. I already had that idea. But reading the [manual](https://doc.qt.io/qt-6/qlist.html) I found that I should give a max size: QList<QString> list(200);. But I didn't want to set up a limit (what would be a good value?).

Reply:
@sunderme both resolved by latest update